### ISSUE TITLE
Declarative CSS Modules and Declarative Shadow DOM `adoptedstylesheets` attribute

### ISSUE THREAD

**Issue by KurtCattiSchmidt** [WebKit/standards-positions#407]
### Title of the proposal

Declarative CSS Modules and Declarative Shadow DOM `adoptedstylesheets` attribute

### URL to the spec

https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/ShadowDOM/explainer.md

### Explainer URL

https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/ShadowDOM/explainer.md

### TAG Design Review URL

TODO

### Mozilla standards-positions issue URL

https://github.com/mozilla/standards-positions/issues/1081

### Description

When developing web components, web authors often encounter challenges with distributing global styles into shadow roots and sharing styles across different shadow roots. [Declarative shadow DOM (DSD)](https://developer.chrome.com/docs/css-ui/declarative-shadow-dom) enables creation of shadow DOM without JS, but adding styles to DSD requires the developer to either use JS to put a shared stylesheet into `adoptedStyleSheets`, or to duplicate the styles in a `<style>` element for each component instance.

We propose an enhancement to CSS module scripts that would allow developers to create CSS module scripts declaratively without a network request, and apply them to DSDs without the use of JS.

**Comment by KurtCattiSchmidt** [comment]
A question specific to WebKit - would your engine support declarative modules with script disabled? Or is your module system tied to the script engine being active? 

I built a prototype in Chromium that works with script disabled but realized this might be an issue in other engines.

**Comment by annevk** [comment]
I don't see any reason that would not work, but note that there's a number of open issues around script disablement in the HTML standard today.

As for this proposal, colleagues and I are generally supportive of solving style sheets for declarative shadow trees (though we'd prefer a more general solution for declarative custom elements), but this proposal isn't far enough along yet to properly evaluate.

**Comment by KurtCattiSchmidt** [comment]
The explainer has been updated recently with a much simpler method for defining this declaratively. Does WebKit have a position on the current proposal? The comments in https://github.com/whatwg/html/issues/10673#issuecomment-3311515023 seem generally favorable.

**Comment by KurtCattiSchmidt** [comment]
Note that the title has been updated and this feature has been scoped to only include `shadowrootadoptedstylesheets`

**Comment by rniwa** [comment]
Is https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/ShadowDOM/explainer.md still the correct description of what's being proposed?

**Comment by KurtCattiSchmidt** [comment]
Yeah, the explainer is up-to-date, specifically this is https://github.com/whatwg/html/pull/12339
